src/main/webapp/app/modeling-assessment-editor/modeling-assessment-editor.component.ts

```diff
@@ -178,13 +185,21 @@ export class ModelingAssessmentEditorComponent implements OnInit, OnDestroy {
         this.canOverride = (this.isAuthorized && isBeforeAssessmentDueDate) || this.isAtLeastInstructor;
     }
 
-    onError(): void {
+    onError(error?: string): void {
+        this.hasErrors = true;
         this.submission = undefined;
         this.modelingExercise = undefined;
         this.result = undefined;
         this.model = undefined;
+        if (!error) {
+            error = 'modelingAssessmentEditor.messages.loadSubmissionFailed';
+        }
         this.jhiAlertService.clear();
-        this.jhiAlertService.error('modelingAssessmentEditor.messages.loadSubmissionFailed');
+        this.jhiAlertService.error(error);
+        setTimeout(() => {
```

Comment:
If we introduce timing, testing becomes harder.
It would be better imo to either redirect right after the page access and show the error OR stay on the side with all elements hidden and showing the error.

Reply:
Good point. I changed it so that the user gets redirected right after accessing the page. The lock limit error message is shown in the exercise dashboard anyway.